core/Memory_Model/Storage/TaskStorageClass.cpp

```diff
@@ -52,4 +58,117 @@ bool TaskStorageClass::PostponeTask(const TaskID& id, const Date& date){
     return result;
   }
   return false;
+}
+
+bool TaskStorageClass::LoadFromFile(const std::string &fileName) {
+  if (fileName == ""){
+    return false;
+  }
+  return DeserializeStorage(fileName);
+}
+
+bool TaskStorageClass::SaveToFile(const std::string &fileName) {
+  if(fileName == ""){
+    return false;
+  }
+  return SerializeStorage(fileName);
+}
+
+bool TaskStorageClass::DeserializeStorage(const std::string& fileName){
+  tasks_.clear();
+  taskIDGenerate_ = TaskIDGenerate(0);
+  SerializedStorage storage;
+  std::ifstream file(fileName);
+
+  if (!file.is_open()){
+    return false;
+  }
+
+  storage.ParseFromIstream(&file);
+  file.close();
+  for (const auto& task : storage.tasks()){
+    auto newTask = Task::Create(task.task().name(),
+                                task.task().label(),
+                                SerializedPriorityToPriority(task.task().priority()),
+                                boost::gregorian::date(task.task().date()));
+
+    if (task.rootid() == task.id()){
+      AddTask(newTask.value());
+    } else {
+      AddSubtask(TaskID(task.rootid()), newTask.value());
+    }
+
+    if (task.complete()) {
+      auto TaskToComplete = GetTask(TaskID(task.id()));
+      TaskToComplete.value()->SetComplete();
+    }
+  }
+  return true;
+
+}
+
+bool TaskStorageClass::SerializeStorage(const std::string& fileName) {
+  //      oldID  newID
+  std::map<TaskID, std::uint32_t> idMapping;
+
+  int i = 0;
+  for (const auto &task : tasks_) {
+    idMapping[task.first] = i;
+    i++;
+  }
+
+  SerializedStorage storageToSave;
+
+  for (const auto& task : tasks_){
+    auto* addTask = storageToSave.add_tasks();
+    auto* newTask = new SerializedTask;
```

Comment:
No good reason for using raw pointer. Should use unique_ptr.

Reply:
Why? This pointer only need to fill the task. Protobuf is responsible to delete it.

Reply:
Protobuf is responible for deleting addTask but not newTask.